proposals/kroxylicious-operator-api-alternative.md

```diff
@@ -498,21 +546,30 @@ spec:
          ...
 
   targetCluster:
-     bootstrapping:
-       bootstrap: upstream:9092
-       protocol: TLS
-       nodeIdRanges:
-       - name: mybrokers
-         range:
-          startInclusive: 0
-          endExclusive: 3
+    clusterRef:
+      name: mytargetcluster
 
   filterRefs:
   - group: filter.kroxylicious.io
     kind: KafkaProtocolFilter
     name: encryption 
 ```
 
+```yaml
+apiVersion: kroxylicious.io/v1alpha1
+kind: KafkaClusterRef
+metadata:
+  name: mytargetcluster
+spec:
+  bootstrap: upstream:9092
+  protocol: TLS
+  nodeIdRanges:
```

Comment:
I'm wondering if `nodeIdRanges` should be here when we might need it for other ref types. Maybe it could be moved up on the `TargetCluster` object. It could be a useful stepping stone for Strimzi integration so that we can extract the addresses from the Kafka CR and not have to go delving for the nodeIds from the relevant KafkaNodePool.

Reply:
If we can deterministically map based on ID's without user input then we should do that. I can see your point @robobario an if we can't do a deterministic & stable mapping without user config then we probably do need to move or duplicate it.

Reply:
We are suck it and see.